apps/perf-test/src/index.scenarios.tsx

```diff
@@ -10,18 +10,57 @@ initializeIcons();
 const div = document.createElement('div');
 document.body.appendChild(div);
 
+const renderFinishedMarkerId = 'render-done';
+const renderFinishedMarker = document.createElement('div');
+renderFinishedMarker.id = renderFinishedMarkerId;
+
 // TODO: could default to displaying list of scenarios if param is not provided.
 const defaultScenarioName = Object.keys(scenarios)[0];
 const defaultIterations = 10;
 
 const queryParams = qs.parse(window.location.search.substring(1));
 const iterations = queryParams.iterations ? parseInt(queryParams.iterations as string, 10) : defaultIterations;
 const scenario = queryParams.scenario ? (queryParams.scenario as string) : defaultScenarioName;
+const renderType = queryParams.renderType;
+
+const PerfTestScenario = scenarios[scenario];
+
+if (renderType === 'rerender') {
+  const PerfTestScenarioWrapper: React.FunctionComponent = () => {
+    const [renderCount, setRenderCounter] = React.useState(0);
+    React.useEffect(() => {
+      if (renderCount < iterations) {
+        setRenderCounter(renderCount + 1);
+      }
+    }, [renderCount]);
```

Comment:
this setup code to trigger re-render seems little costly and it reflects in the flame chart

Reply:
This is only measuring React virtual rendering time with no changes to the actual DOM hierarchy, correct? In other words, this doesn't capture layout and styling time, cost to update browser DOM, etc.

However, when we do a `mount` test, it does wait for the entire DOM hierarchy to be populated by waiting for the page load event. I think this is an important difference to call out.

If we wanted to measure the cost of layout and actual DOM updates, I could see a similar approach to watching for the `render-done` marker with a counter perhaps, and then rendering again. Although it seems a bit hacky and wonder if there are better approaches.

Reply:
We also chatted about this but just to document: The fluentui `perf-test` is a "newer" version of this file, and also uses `digest` to create a bundle of test scenarios. I always thought it'd be relatively easy to change fabric to use fluentui's `perf-test` package as it has been.

However, now that improvements are also being made to this file, it will be harder to converge the perf-test packages. We should also make sure these changes will make sense in the context of fluentui `perf-test` and `digest`.

Reply:
as discussed offline. I now calling the new render type "virtual re-render". also added comments in `scenarioRenderTypes` to clarify when such test is most useful.

I complete agree with the concern on having 2 perf-test scripts, I would love to make progress on that when i find chance. Right now given my priorities, I don't have time to do that ATM.